### PR TITLE
Chmap mixer support

### PR DESCRIPTION
**Pull request by rohit-qc**
There can be multiple chmap mixer control in a platform (e.g. - Playback Channel map / Capture Channel map) for each pcm devices. Tinyalsa does not provide an api to get the mixer_ctl handle based on combination of mixer control name and pcm_device id. This adds support for getting mixer_ctl handle for such cases.

### CODE REVIEW

File: include/tinyalsa/mixer.h [hunk 1]
```diff
@@ -93,6 +93,8 @@ int mixer_add_new_ctls(struct mixer *mixer);
 
 const char *mixer_get_name(const struct mixer *mixer);
 
+unsigned int mixer_ctl_get_dev_num(const struct mixer_ctl *ctl);
```
Comment (dvdli) [inline include/tinyalsa/mixer.h:96]: 1. Please move this function declaration to the lines between 120 and 160.
2. Please rename to `mixer_ctl_get_device` to match the naming rule.

File: include/tinyalsa/mixer.h [hunk 2]
```diff
@@ -103,6 +105,10 @@ struct mixer_ctl *mixer_get_ctl(struct mixer *mixer, unsigned int id);
 
 struct mixer_ctl *mixer_get_ctl_by_name(struct mixer *mixer, const char *name);
 
+struct mixer_ctl *mixer_get_ctl_by_name_and_device(struct mixer *mixer,
+                                                  const char *name,
+                                                  unsigned int device);
```
Comment (dvdli) [inline include/tinyalsa/mixer.h:110]: Add one space for alignment.

File: src/mixer.c [hunk 3]
```diff
@@ -760,6 +760,51 @@ struct mixer_ctl *mixer_get_ctl_by_name_and_index(struct mixer *mixer,
     return NULL;
 }
 
+/** Gets an instance of mixer control handle, by the mixer control's name and device.
+ *  For instance, if two controls have same name, e.g. 'Playback Channel map', then PCM device returns the specific control.
```
Comment (dvdli) [inline src/mixer.c:764]: For better readability, the column limit is 100 chars. Please help to make sure all lines don't exceed.

File: src/mixer.c [hunk 4]
```diff
@@ -760,6 +760,51 @@ struct mixer_ctl *mixer_get_ctl_by_name_and_index(struct mixer *mixer,
...
+        }
+    }
+
+#ifdef TINYALSA_USES_PLUGINS
+    if (mixer->v_grp) {
+        grp = mixer->v_grp;
+        ctl = grp->ctl;
+
+        for (n = 0; n < grp->count; n++)
+            if ((!strcmp(name, (char*) ctl[n].info.id.name)) && (device == ctl[n].info.id.device)) {
```
Comment (dvdli) [inline src/mixer.c:800]: No need to add one more pair of parentheses. This is also for line 788.

```
if (!strcmp(name, (char*) ctl[n].info.id.name) &&
        device == ctl[n].info.id.device) {
```

**Review by dvdli — CHANGES_REQUESTED** [review]

File: include/tinyalsa/mixer.h (hunk 1 above)
Comment (rohit-qc) [reply include/tinyalsa/mixer.h]: Thanks for the review. Updated changes in pull request.

File: include/tinyalsa/mixer.h (hunk 2 above)
Comment (rohit-qc) [reply include/tinyalsa/mixer.h]: done

File: src/mixer.c (hunk 3 above)
Comment (rohit-qc) [reply src/mixer.c]: updated

File: src/mixer.c (hunk 4 above)
Comment (rohit-qc) [reply src/mixer.c]: fixed at both places

File: src/mixer.c [hunk 5]
```diff
@@ -760,6 +760,54 @@ struct mixer_ctl *mixer_get_ctl_by_name_and_index(struct mixer *mixer,
...
+ *  For instance, if two controls have same name,
+ *  e.g. 'Playback Channel map', then PCM device returns the specific control.
+ * @param mixer An initialized mixer handle.
+ * @param name The control's name in the given mixer.
+ * @param device The PCM device
+ * @returns A handle to the mixer control.
+ * @ingroup libtinyalsa-mixer
+ */
+struct mixer_ctl *mixer_get_ctl_by_name_and_device(struct mixer *mixer,
+                                                  const char *name,
```
Comment (dvdli) [inline src/mixer.c:773]: Please also help to add a space to line 773 and 774. Thanks.

File: src/mixer.c [hunk 6]
```diff
@@ -760,6 +760,54 @@ struct mixer_ctl *mixer_get_ctl_by_name_and_index(struct mixer *mixer,
...
+#ifdef TINYALSA_USES_PLUGINS
+    if (mixer->v_grp) {
+        grp = mixer->v_grp;
+        ctl = grp->ctl;
+
+        for (n = 0; n < grp->count; n++)
+            if (!strcmp(name, (char*) ctl[n].info.id.name) &&
+                    device == ctl[n].info.id.device) {
+                return ctl + n;
+            }
```
Comment (dvdli) [inline src/mixer.c:805]: Please also add a pair of brackets for the `for` loop.

**Review by dvdli — CHANGES_REQUESTED** [review]

File: src/mixer.c (hunk 6 above)
Comment (rohit-qc) [reply src/mixer.c:805]: Done

File: src/mixer.c (hunk 5 above)
Comment (rohit-qc) [reply src/mixer.c]: Sorry that I missed updating here. Updated now.

**Review by dvdli — APPROVED** [review]
LGTM

**Pull request merged**